app/src/main/java/com/keylesspalace/tusky/components/compose/ComposeActivity.kt

```diff
@@ -707,38 +698,25 @@ class ComposeActivity : BaseActivity(),
     }
 
     /** This is for the fancy keyboards which can insert images and stuff. */
-    override fun onCommitContent(inputContentInfo: InputContentInfoCompat, flags: Int, opts: Bundle): Boolean {
-        try {
-            currentInputContentInfo?.releasePermission()
-        } catch (e: Exception) {
-            Log.e(TAG, "InputContentInfoCompat#releasePermission() failed." + e.message)
-        } finally {
-            currentInputContentInfo = null
-        }
-
+    override fun onCommitContent(inputContentInfo: InputContentInfoCompat, flags: Int, opts: Bundle?): Boolean {
         // Verify the returned content's type is of the correct MIME type
         val supported = inputContentInfo.description.hasMimeType("image/*")
 
-        return supported && onCommitContentInternal(inputContentInfo, flags)
-    }
-
-    private fun onCommitContentInternal(inputContentInfo: InputContentInfoCompat, flags: Int): Boolean {
-        if (flags and InputConnectionCompat.INPUT_CONTENT_GRANT_READ_URI_PERMISSION != 0) {
-            try {
-                inputContentInfo.requestPermission()
-            } catch (e: Exception) {
-                Log.e(TAG, "InputContentInfoCompat#requestPermission() failed." + e.message)
-                return false
+        if(supported) {
+            val lacksPermission = (flags and InputConnectionCompat.INPUT_CONTENT_GRANT_READ_URI_PERMISSION) != 0
```

Comment:
this isn't new but I'm really confused about this. Isn't flag check
`flags & myFlag == myFlag`?
https://gist.github.com/LouisCAD/04be811e22d45cbc7285ab247c6c12e7#file-bitflags-kt-L7

and here we then want to check that it's not there then it should be `0` or?..

Reply:
this is how its done in the documentation as well & it works